satpy/etc/readers/viirs_edr_active_fires.yaml

```diff
@@ -5,44 +5,64 @@ reader:
     sensors: [viirs]
 
 file_types:
+    fires_netcdf_img:
+        file_reader: !!python/name:satpy.readers.viirs_edr_active_fires.VIIRSActiveFiresFileHandler
+        variable_prefix: ""
+        file_patterns:
+            - 'AFIMG_npp_d{start_time:%Y%m%d_t%H%M%S%f}_e{end_time:%H%M%S%f}_b{orbit:5d}_c{creation_time}_{source}.nc'
     fires_netcdf:
         file_reader: !!python/name:satpy.readers.viirs_edr_active_fires.VIIRSActiveFiresFileHandler
+        variable_prefix: "Fire Pixels/"
         file_patterns:
-            - 'AFEDR_npp_d{start_time:%Y%m%d_t%H%M%S%f}_e{end_time:%H%M%S%f}_b{orbit:5d}_c{creation_time}_{source}.nc'
+        - 'AFEDR_npp_d{start_time:%Y%m%d_t%H%M%S%f}_e{end_time:%H%M%S%f}_b{orbit:5d}_c{creation_time}_{source}.nc'
+        - 'AFMOD_npp_d{start_time:%Y%m%d_t%H%M%S%f}_e{end_time:%H%M%S%f}_b{orbit:5d}_c{creation_time}_{source}.nc'
     fires_text:
         file_reader: !!python/name:satpy.readers.viirs_edr_active_fires.VIIRSActiveFiresTextFileHandler
         file_patterns:
-            - 'AFEDR_npp_d{start_time:%Y%m%d_t%H%M%S%f}_e{end_time:%H%M%S%f}_b{orbit:5d}_c{creation_time}_{source}.txt'
+            - '{satellite_name}_npp_d{start_time:%Y%m%d_t%H%M%S%f}_e{end_time:%H%M%S%f}_b{orbit:5d}_c{creation_time}_{source}.txt'
 
 datasets:
-    detection_confidence:
-        name: detection_confidence
-        standard_name: fire_confidence
+    confidence_cat:
+        name: confidence_cat
+        standard_name: confidence_cat
+        file_type: [fires_text, fires_netcdf_img]
```

Comment:
How can both fire confidences be in `fires_text`? Additionally the NetCDF file type should be listed before the text file type.

Reply:
I use one parser for both MOD and IMG filetypes. Depending on the satellite_name attribute, I read the CSV into a dataset with the relevant headers

Do you think it's better to separate them into two different readers?

Reply:
If someone gives you a MOD and IMG text file, then Satpy will see them as two files that *both* need to be read to get the confidence data. This would result in a concatenation of the data (most likely) which is not what we want. Additionally, isn't the confidence in one of the text files percent and the other is category?

Also, the order needs to be swapped regardless, netcdf should come before text.

Unless you can point out why I'm wrong, I would separate the file types (the python code can still be one class).